### ISSUE TITLE
CDO fails to build on Hera

### ISSUE THREAD

**Issue by kgerheiser** [JCSDA/spack-stack#248]
Using Intel 2022. 

Of course the CDO developers redirect stdout/stderr `/dev/null 2&1` so there's no good error message.

**Comment by kgerheiser** [comment]
The redirect to `/dev/null` is done by `libtool` when building the PIC version of files because the output should be identical to the non-PIC build of the same file.

The real error is an internal compiler error when building the PIC variant of `fieldc_complex.cc`

`compare.h(13) (col. 45): internal error: 04010002_12261`

Which is this simple function:

```
//#define DBL_IS_EQUAL(x,y) (!(x < y || y < x))
//#define DBL_IS_EQUAL(x, y) (std::isnan(x) || std::isnan(y) ? (std::isnan(x) && std::isnan(y) ? 1 : 0) : !(x < y || y < x))
static inline bool DBL_IS_EQUAL(double x, double y)
{
  return (std::isnan(x) || std::isnan(y) ? (std::isnan(x) && std::isnan(y)) : !(x < y || y < x));
}
```

Commenting out this function and un-commenting `//#define DBL_IS_EQUAL(x, y) (std::isnan(x) || std::isnan(y) ? (std::isnan(x) && std::isnan(y) ? 1 : 0) : !(x < y || y < x))`

And no more error. This isn't an ideal workaround to use within Spack.

Strange that it only shows up when building the PIC variant, not in the CI using Intel (though it could've been fixed since 2022.1.2 version on Hera).

CDO has a configure option `--with-pic=yes/no` that I'm going to try and set to `no`.

**Comment by kgerheiser** [comment]
Manually expanding the ternary operator seems to satisfy the compiler:

```
--- compare.h	2021-10-29 12:00:30.000000000 -0400
+++ compare.h.new	2022-07-11 17:04:31.000000000 -0400
@@ -10,7 +10,11 @@
 static inline bool
 DBL_IS_EQUAL(double x, double y)
 {
-  return (std::isnan(x) || std::isnan(y) ? (std::isnan(x) && std::isnan(y)) : !(x < y || y < x));
+    if (std::isnan(x) || std::isnan(y)) {
+        return std::isnan(x) && std::isnan(y);
+    } else {
+        return !(x < y || y < x);
+    }
 }
 
 //#define IS_NOT_EQUAL(x, y) (x < y || y < x)
}
